File: OrderManagement/MultiProjectSample/Models/Models/OrderDetailsModel.cs
namespace MultiProjectSample.Models.Models
{
    public class OrderDetailsModel : BaseModel
    {
        public int OrderID { get; set; }
        public int ProductID { get; set; }
        public decimal UnitPrice { get; set; }
        public short Quantity { get; set; }
        public float Discount { get; set; }
        public virtual ProductModel Product { get; set; }
    }
}

File: OrderManagement/MultiProjectSample/app/models/order.details.model.ts
import { BaseModel } from "../models/base.model";

export class OrderDetailModel extends BaseModel  {
    orderID: number;
    productID: number;
    unitPrice: number;
    quantity: number;
    discount: number;
    product: any;

    constructor(orderDetailData: any) {
        super();

        this.orderID = super.checkUndefinedValue(orderDetailData.orderID);
        this.productID = super.checkUndefinedValue(orderDetailData.productID);
        this.unitPrice = super.checkUndefinedValue(orderDetailData.unitPrice);
        this.quantity = super.checkUndefinedValue(orderDetailData.quantity);
        this.discount = super.checkUndefinedValue(orderDetailData.discount);
        this.product = orderDetailData.product;
    }
}